### PR TITLE
allow keys with internal nul chars in rust bindings

### PR DESCRIPTION
**Pull request by gmossessian**
Due to difference between Rust string and C string, it is possible to cause panic by submitting a string with a nul character in the middle to dictionary::get. 

This PR returns empty match instead.

### CODE REVIEW

File: rust/Cargo.toml
Comment (hendrikmuhs) [inline rust/Cargo.toml:3]: this shouldn't be part of this PR. 

We bump versions as part of a release, there are some more things to do for that (branching, tagging, ...)

Having that said, we can create a patch release soonish if you need the fix.

**Review by hendrikmuhs — COMMENTED** [review]
@narekgharibyan is the maintainer of the C and rust binding.

However, I have reservations regarding the approach. The limitation is a bug in the rust binding, keyvi supports zero bytes. Returning an empty match as proposed silences a bug. I would understand if the rust binding errors in some way and this error is handled on client side without causing `panic`.

Would it be possible to provide an additional get method in the C-API that takes a pointer and a length? I saw a container called `keyvi_bytes` in the code used in a different context but basically implementing this approach.

**Comment by narekgharibyan** [comment]
> Therefore I leave it up to you.

Then I'd say no need for a new one, and we can just change existing one.

>That means string and binary handling became much easier and we could actually add support for byte objects, if we have a usecase for it.

Cool, then let's keep rust API part as is for now, and change it in the future if need be.

File: rust/Cargo.toml
Comment (gmossessian) [reply rust/Cargo.toml]: thank you, removed version bump in 5dec183

**Comment by gmossessian** [comment]
Thanks for the discussion @hendrikmuhs and @narekgharibyan, I agree this is a better approach. I'll take a look at the library and re-request review when it's ready :)

@nararekgharibyan `cargo fmt` does not return any errors for me locally, furthermore I cannot see CI, I'm not sure it can be run on this PR without approval as I am a first-time contributor.

File: keyvi/bin/keyvi_c/c_api.cpp
Comment (hendrikmuhs) [inline keyvi/bin/keyvi_c/c_api.cpp:112]: now that I see the API, I realize we don't need to wrap and unwrap it:

```
keyvi_match* keyvi_dictionary_get(const keyvi_dictionary* dict, const char* key, size_t length)
```

**Review by hendrikmuhs — COMMENTED** [review]
sorry, it was my suggestion to use keyvi_bytes, actually we don't need it.

(FWIW: its odd that we still create a copy, because the the API requires a `std::string`, but lets ignore that for now, we can switch the dictionary getter to a `string_view`, but that requires C++17)

File: keyvi/bin/keyvi_c/c_api.cpp
Comment (gmossessian) [reply keyvi/bin/keyvi_c/c_api.cpp]: good point! done in 6073fdd

**Review by hendrikmuhs — APPROVED** [review]
Thanks! LGTM

**Comment by gmossessian** [comment]
Thank you @hendrikmuhs! I'll wait for @narekgharibyan's final approval as well.

**Review by narekgharibyan — APPROVED** [review]
@gmossessian thanks for fixing it. LGTM!

**Pull request merged**
